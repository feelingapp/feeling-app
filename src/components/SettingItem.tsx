import { styled } from "linaria/react"

const Container = styled.a``

const Title = styled.div`
  font-size: 1.8rem;
  font-weight: 500;
  margin-bottom: var(--space-xxxs);
`

const Subtitle = styled.div`
  color: var(--grey-color);
  font-size: 1.4rem;
`

interface Props {
  title: string
  subtitle?: string
  href?: string
}

export default function SettingItem(props: Props): JSX.Element {
  const { title, subtitle, href } = props

  return (
    <Container as={href ? "a" : "div"} href={href}>
      <Title>{title}</Title>
      {subtitle && <Subtitle>{subtitle}</Subtitle>}
    </Container>
  )
}
