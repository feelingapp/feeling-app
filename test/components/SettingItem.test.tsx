import { shallow } from "enzyme"
import SettingItem from "../../src/components/SettingItem"

describe("Setting: ", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(
      <SettingItem title="Lorem ipsum" subtitle="dolor sit amet" href="/" />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("matches snapshot without subtitle", () => {
    const wrapper = shallow(<SettingItem title="Lorem ipsum" href="/" />)
    expect(wrapper).toMatchSnapshot()
  })

  it("matches snapshot without href", () => {
    const wrapper = shallow(
      <SettingItem title="Lorem ipsum" subtitle="dolor sit amet" />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("matches snapshot without subtitle and href", () => {
    const wrapper = shallow(<SettingItem title="Lorem ipsum" />)
    expect(wrapper).toMatchSnapshot()
  })
})
