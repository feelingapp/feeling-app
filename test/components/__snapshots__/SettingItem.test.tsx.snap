// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting:  matches snapshot 1`] = `
<Container
  as="a"
  href="/"
>
  <Title>
    Lorem ipsum
  </Title>
  <Subtitle>
    dolor sit amet
  </Subtitle>
</Container>
`;

exports[`Setting:  matches snapshot without href 1`] = `
<Container
  as="div"
>
  <Title>
    Lorem ipsum
  </Title>
  <Subtitle>
    dolor sit amet
  </Subtitle>
</Container>
`;

exports[`Setting:  matches snapshot without subtitle 1`] = `
<Container
  as="a"
  href="/"
>
  <Title>
    Lorem ipsum
  </Title>
</Container>
`;

exports[`Setting:  matches snapshot without subtitle and href 1`] = `
<Container
  as="div"
>
  <Title>
    Lorem ipsum
  </Title>
</Container>
`;
